### ISSUE TITLE
Replication of Maverick training

### ISSUE THREAD

**Issue by LeonidDybowski** [ZuchnerLab/Maverick#15]
Dear Developers,

Maverick is a great tool, thank you for such an interesting model. 
I am trying to repeat the training with the same data, but unfortunately, it hasn't worked yet. The model always predicts benign class for me. The only thing that I changed were a few rows in the `DataGenerator` class. I moved the assignment of `transcriptID`, `changePos`, `AltSeq` and `WTSeq` variables outside of the `self.returnStyle==2` condition, because otherwise they were not assigned during data generation for Architecture 1.
I have tried to change the learning rate a little bit, but it did not help: the Architecture 1 still predicts benign class always, even though there are pathogenic labels in the data. The training history shows me `loss: nan - accuracy: 0.7904` for all 20 epochs.
Maybe you know what could be the reason behind this?

Thank you.

**Comment by mdanzi** [comment]
Hello, 

Thanks for your interest in using Maverick! Sorry that you are running into issues replicating the training. Could you attach your new code for the DataGenerator? I'm guessing you are working off the TrainMaverick.ipynb file as the 'self.returnStyle' nomenclature was dropped a while ago from the inference code. Try using the code for the DataGenerator that exists in the InferenceScripts/runModels.py file and see if that helps too. 

Best,
Matt

**Comment by mdanzi** [comment]
Hello Leonid,

Sorry for my slow reply! Thank you for uploading that code. That is quite odd. I am still not sure why the code you sent is not working. But, yes, I think the easiest thing for you to do would be to use the DataGenerator class from the runModels.py file to train both architectures. But you may need to still use the 'returnStyle' parameter to adjust the output like I do in lines 651-652 of runModels.py so that you still get the correct dictionaries passed to each architecture. Initially, I thought you could use dictionaries with extra keys and those extra bits would just get ignored, but now I think that was incorrect (I was thinking of something in pytorch, not tensorflow). 

Anyway, I would suggest something like this for the DataGenerator (working off the runModels.py one):
```
class DataGenerator(keras.utils.Sequence):
    def __init__(self, list_IDs, labels, dataFrameIn, tokenizer, T5Model, batch_size=32, padding=100, n_channels_emb=1024, n_channels_mm=51, n_classes=3, shuffle=True, returnStyle=1):
        self.returnStyle=returnStyle
    ... lines 132 through 545 kept the same ...
        X={'mm_orig_seq':mm_orig,'mm_alt_seq':mm_alt,'non_seq_info':nonSeq}
        if self.returnStyle==2:
            X={'alt_cons':mm_alt,'alt_emb':altEmbeddings,'non_seq_info':nonSeq}
        return X, keras.utils.to_categorical(y, num_classes=self.n_classes)
```

Then hopefully you can call this DataGenerator the same way you did in the code you sent earlier. Let me know if it works!

**Comment by LeonidDybovskij** [comment]
Dear Matt,

Thank you for the detailed reply!
I replaced the code with the one from runModels.py, and the model trains now! 
At least Architecture 1. The second one I have not tested yet, but I think it will work as well.

Thanks again,
Leonid
